# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1520605338 18000
#      Fri Mar 09 09:22:18 2018 -0500
# Node ID 1e207bd05b5d35c64c96141c5809747aee6bbde5
# Parent  53db3b152397a686c0d2f124452ca23ef3d872f6
avoid AWT test on OS X (bug #53305)

* ov-java.cc: Don't attempt to use AWT in test on OS X systems.

diff --git a/libinterp/octave-value/ov-java.cc b/libinterp/octave-value/ov-java.cc
--- a/libinterp/octave-value/ov-java.cc
+++ b/libinterp/octave-value/ov-java.cc
@@ -3427,13 +3427,16 @@ Return true if @var{x} is a Java object.
 %! assert (! a(1).s.isEmpty ())
 %! assert (a(1).s.length (), 6)
 %! assert (a(1).s.charAt (0), "O")
 %! assert (a(1).s.charAt (5), "e")
 %! assert (a(1).s.matches ("^Octave$"))
 %! assert (a(1).s.startsWith ("Oct"))
 
 ## Check for basic usability of the java awt library
-%!testif HAVE_JAVA; usejava ("jvm") && usejava ("awt") && have_window_system ()
+## Skip the test on OS X where we currently have Java 9 and attempting
+## to use awt causes Octave to exit with a message about Java not being
+## installed (it is) instead of returning false.
+%!testif HAVE_JAVA; ! ismac () && usejava ("jvm") && usejava ("awt") && have_window_system ()
 %! frame = javaObject ("java.awt.Frame");
 %! frame.setResizable (true);
 %! assert (frame.isResizable ());
 */
